### PR TITLE
Mark generate_tests as deprecated and lint for it

### PR DESCRIPTION
**Pull request by gsnedders**
This is the start of killing generate_tests.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/6735)

### CODE REVIEW

File: docs/_writing-tests/testharness-api.md [hunk 1]
```diff
@@ -451,8 +451,14 @@ generate_tests(assert_equals, [
 Is equivalent to:
 
 ```js
-test(function() {assert_equals(1+1, 2)}, "Sum one and one")
-test(function() {assert_equals(1+0, 1)}, "Sum one and zero")
+var a_got = 1+1;
```
Comment (jgraham) [inline docs/_writing-tests/testharness-api.md]: Why this change?

**Review by jgraham — APPROVED** [review]

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260667335)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260667336)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260667344)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260667345)

# Sauce (MicrosoftEdge) #

File: docs/_writing-tests/testharness-api.md (hunk 1 above)
Comment (gsnedders) [inline docs/_writing-tests/testharness-api.md]: To better demonstrate the shortcoming of `generate_tests`.

**Review by sideshowbarker — COMMENTED** [review]
LGTM
